library/core/src/slice/ascii.rs

```diff
@@ -27,6 +27,15 @@ impl [u8] {
         self.len() == other.len() && iter::zip(self, other).all(|(a, b)| a.eq_ignore_ascii_case(b))
     }
 
+    /// Checks if all bytes are ASCII decimal digit
+    #[inline]
+    pub fn is_ascii_digit(&self) -> bool {
```

Comment:
This needs stability attributes, but can wait until after you do an ACP for the change.